include/cppkafka/utils/buffered_producer.h

```diff
@@ -163,9 +189,11 @@ class CPPKAFKA_API BufferedProducer {
 
     Producer producer_;
     QueueType messages_;
+    mutable std::mutex mutex_;
     ProduceFailureCallback produce_failure_callback_;
-    size_t expected_acks_{0};
-    size_t messages_acked_{0};
+    std::atomic_ulong expected_acks_{0};
+    std::atomic_ullong total_messages_acked_{0};
+    std::atomic_ushort rollover_counter_{0};
```

Comment:
Is this really necessary? I understand that if you want to be "correct" this is the right way to go. If you explicitly use something like `atomic<uint64_t>` for the total messages acked member, you know there's absolutely no way you would *ever* produce that amount of messages and this would remove the need to have this counter.

Reply:
I will remove it. I guess I was trying to be correct rather than practical :)